### PR TITLE
HW-66780 : [iOS] Add List prepaid card balances to Core SDK

### PR DESCRIPTION
**Pull request by ddhar-hw**

### CODE REVIEW

File: Tests/Balance/HyperwalletPrepaidCardBalanceTests.swift [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+        Hyperwallet.setup(HyperwalletTestHelper.authenticationProvider)
+    }
+
+    override func tearDown() {
+        if Hippolyte.shared.isStarted {
+            Hippolyte.shared.stop()
+        }
+    }
+
+    func testListUserBalances_success() {
```
Comment (jsaini-hw) [inline Tests/Balance/HyperwalletPrepaidCardBalanceTests.swift:24]: rename method name to `testListPrepaidCardBalances_success`

File: Tests/Balance/HyperwalletPrepaidCardBalanceTests.swift [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+        if testCaseDescription != "Empty Result" {
+            // Given
+            let expectation = self.expectation(description: "List Prepaid Card Balances completed")
+            let response = HyperwalletTestHelper.okHTTPResponse(for: mockResponseFileName!)
+            let url = String(format: "%@/prepaid-cards/%@/balances", HyperwalletTestHelper.userRestURL, "trm-1234")
+            let request = HyperwalletTestHelper.buildGetRequestRegexMatcher(pattern: url, response)
+            HyperwalletTestHelper.setUpMockServer(request: request)
+
+            var userBalanceList: HyperwalletPageList<HyperwalletBalance>?
```
Comment (jsaini-hw) [inline Tests/Balance/HyperwalletPrepaidCardBalanceTests.swift:33]: prepaidCardBalanceList

File: Tests/Balance/HyperwalletPrepaidCardBalanceTests.swift [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+            var errorResponse: HyperwalletErrorType?
+
+            // When
+            var balanceQueryParam: HyperwalletPrepaidCardBalanceQueryParam?
+            if sortBy != nil || currency != nil {
```
Comment (jsaini-hw) [inline Tests/Balance/HyperwalletPrepaidCardBalanceTests.swift:38]: No need to check currency

File: Tests/Balance/HyperwalletPrepaidCardBalanceTests.swift [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+            if let userBalance = userBalanceList?.data?.first {
+                XCTAssertEqual(userBalance.amount, amount)
+                XCTAssertEqual(userBalance.currency, expectedCurrency)
+            } else {
+                assertionFailure("\(testCaseDescription!) - The user balance should be not nil")
+            }
+        }
+    }
+
+    func testListUserBalances_emptyResult() {
```
Comment (jsaini-hw) [inline Tests/Balance/HyperwalletPrepaidCardBalanceTests.swift:70]: update method name

File: Tests/Balance/HyperwalletPrepaidCardBalanceTests.swift [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+        if testCaseDescription == "Empty Result" {
+            // Given
+            let expectation = self.expectation(description: "List Prepaid Card Balances completed")
+            let response = HyperwalletTestHelper.noContentHTTPResponse()
+            let url = String(format: "%@/prepaid-cards/%@/balances", HyperwalletTestHelper.userRestURL, "trm-1234")
+            let request = HyperwalletTestHelper.buildGetRequestRegexMatcher(pattern: url, response)
+            HyperwalletTestHelper.setUpMockServer(request: request)
+
+            var userBalanceList: HyperwalletPageList<HyperwalletBalance>?
```
Comment (jsaini-hw) [inline Tests/Balance/HyperwalletPrepaidCardBalanceTests.swift:79]: update var name

**Review by jsaini-hw — COMMENTED** [review]
Update Readme.md

**Review by jsaini-hw — APPROVED** [review]

**Review by malaw-hw — APPROVED** [review]

**Pull request merged**